tasks/libs/common/utils.py

```diff
@@ -54,6 +54,13 @@ def is_allowed_repo_nightly_branch(branch):
     return branch in ALLOWED_REPO_NIGHTLY_BRANCHES
 
 
+def on_release_context(bucket):
+    """
+    Util for specific computation of the last valid tag, to prevent interference with custom tags.
+    """
+    return os.getenv("RELEASE_CONTEXT") == "true" or is_allowed_repo_nightly_branch(bucket)
```

Comment:
If you really want to go with an env variable instead of a new parameter, I would: 

- Not set the env variable in the Python code itself anywhere and configure it directly in the workflow [here](https://github.com/DataDog/datadog-agent/blob/a7b80d61ea5d49073442976329a44bd53b2840ab/.github/workflows/create_rc_pr.yml#L57). Example can be found [here](https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow)
- Remove the `bucket` param from the signature of this function because if callers don't need to know about `RELEASE_CONTEXT`, then they should not bother with `BUCKET_BRANCH` either

Otherwise I'd go with a new option that would be used in the GH action. I don't have enough context to provide a better alternative

Reply:
I'm adding a param everywhere, it's less ugly